enterprise/internal/authz/github/github.go

```diff
@@ -134,27 +134,46 @@ func (p *Provider) fetchUserPermsByToken(ctx context.Context, accountID extsvc.A
 	// 100 matches the maximum page size, thus a good default to avoid multiple allocations
 	// when appending the first 100 results to the slice.
 	const repoSetSize = 100
-	perms := &authz.ExternalUserPermissions{
-		Exacts: make([]extsvc.RepoID, 0, repoSetSize),
-	}
-	seenRepos := make(map[extsvc.RepoID]struct{}, repoSetSize)
-
-	// addRepoToUserPerms checks if the given repos are already tracked before adding it to perms.
-	addRepoToUserPerms := func(repos ...extsvc.RepoID) {
-		for _, repo := range repos {
-			if _, exists := seenRepos[repo]; !exists {
-				seenRepos[repo] = struct{}{}
-				perms.Exacts = append(perms.Exacts, repo)
-			}
-		}
-	}
 
-	// If groups caching is enabled, we sync just a subset of direct affiliations - we let
-	// other permissions ('organization' affiliation) be sync'd by teams/orgs.
-	affiliations := []github.RepositoryAffiliation{github.AffiliationOwner, github.AffiliationCollaborator}
-	if p.groupsCache == nil {
-		// Otherwise, sync all direct affiliations.
+	var (
+		// perms tracks repos this user has access to
+		perms = &authz.ExternalUserPermissions{
+			Exacts: make([]extsvc.RepoID, 0, repoSetSize),
```

Comment:
```suggestion
			Exacts: make([]extsvc.RepoID, 0, repoSetSize),
```

Shouldn't we leave it nil and only initialize it when group cache is enabled?

Reply:
This is used regardless of groupsCache (it's the return type), and in the old code (before groupCache) this was instantiated beforehand as well